src/joystick/windows/SDL_dinputjoystick.c

```diff
@@ -236,69 +234,99 @@ SetDIerror(const char *function, HRESULT code)
 }
 
 static SDL_bool
-SDL_IsXInputDevice(const TCHAR *name, const GUID* pGuidProductFromDirectInput)
+SDL_IsXInputDevice(Uint16 vendor_id, Uint16 product_id, const char* hidPath)
 {
-    UINT i;
-
     if (!SDL_XINPUT_Enabled()) {
         return SDL_FALSE;
     }
 
-    if (SDL_tcsstr(name, TEXT(" XINPUT ")) != NULL) {
-        /* This is a duplicate interface for a controller that will show up with XInput,
-           e.g. Xbox One Elite Series 2 in Bluetooth mode.
-         */
+    SDL_GameControllerType type = SDL_GetJoystickGameControllerType("", vendor_id, product_id, -1, 0, 0, 0);
```

Comment:
This hits `-Werror=declaration-after-statement`:
```
src/joystick/windows/SDL_dinputjoystick.c: In function 'SDL_IsXInputDevice':
src/joystick/windows/SDL_dinputjoystick.c:243:5: error: ISO C90 forbids mixed declarations and code
```
The following simple patch makes it c90-compatible and build without errors:
```diff
--- SDL_dinputjoystick.c~
+++ SDL_dinputjoystick.c
@@ -236,11 +236,13 @@ SetDIerror(const char *function, HRESULT
 static SDL_bool
 SDL_IsXInputDevice(Uint16 vendor_id, Uint16 product_id, const char* hidPath)
 {
+    SDL_GameControllerType type;
+
     if (!SDL_XINPUT_Enabled()) {
         return SDL_FALSE;
     }
 
-    SDL_GameControllerType type = SDL_GetJoystickGameControllerType("", vendor_id, product_id, -1, 0, 0, 0);
+    type = SDL_GetJoystickGameControllerType("", vendor_id, product_id, -1, 0, 0, 0);
     if (type == SDL_CONTROLLER_TYPE_XBOX360 ||
         type == SDL_CONTROLLER_TYPE_XBOXONE ||
         (vendor_id == 0x28DE && product_id == 0x11FF)) {
```

Reply:
fixed. thanks!